django/test/client.py

```diff
@@ -551,6 +551,10 @@ def generic(
             (key.lower().encode('ascii'), value.encode('latin1'))
             for key, value in extra.items()
         ]
+
+        if query_string := extra.pop('QUERY_STRING', None):
+            s['query_string'] = query_string
+
```

Comment:
We should do this before setting `headers` because we don't want to have `query_string` in them:
```python
        if query_string := extra.pop('QUERY_STRING', None):
            s['query_string'] = query_string
        s['headers'] += [
             (key.lower().encode('ascii'), value.encode('latin1')) 
             for key, value in extra.items()
        ]
        ...
```

I would also add a test, e.g.
```diff
diff --git a/tests/test_client/tests.py b/tests/test_client/tests.py
index b31a431c01..a32a56b5c9 100644
--- a/tests/test_client/tests.py
+++ b/tests/test_client/tests.py
@@ -1068,3 +1068,8 @@ class AsyncRequestFactoryTest(SimpleTestCase):
         self.assertIn('HTTP_AUTHORIZATION', request.META)
         self.assertEqual(request.headers['x-another-header'], 'some other value')
         self.assertIn('HTTP_X_ANOTHER_HEADER', request.META)
+
+    def test_request_factory_query_string(self):
+        request = self.request_factory.get('/somewhere/', {'example': 'data'})
+        self.assertNotIn('Query-String', request.headers)
+        self.assertEqual(request.GET['example'], 'data')
```

Reply:
Thanks, changes applied.